lightning/src/offers/offer.rs

```diff
@@ -94,6 +97,7 @@ use std::time::SystemTime;
 /// [module-level documentation]: self
 pub struct OfferBuilder {
 	offer: OfferContents,
+	metadata: Metadata,
```

Comment:
It would be nice if we could either unify or disambiguate this `metadata` field with `OfferContents::metadata`, similar for `InvoiceRequest`. At this point, it seems like a more accurate name for this would be `metadata_metadata`?

Reply:
> It would be nice if we could either unify or disambiguate this `metadata` field with `OfferContents::metadata`, similar for `InvoiceRequest`.

I took a stab at this in 7c48b4bb531290d779a5477b26c2fe0c73438a4e and 699550b771058de44bc065da02ac1fda17aaaa51 fixups. The primary issue in my mind is it makes it such that a constructed `Offer` must only use the `Metadata::Bytes` variant. So there are some assumptions that this is the case and an `unwrap` when verifying which otherwise would not be needed. Let me know what you and @TheBlueMatt think before I try the same for `InvoiceRequest` and `Refund`.

> At this point, it seems like a more accurate name for this would be `metadata_metadata`?

Not sure I follow this naming. It's more like metadata construction strategy or the like.

Reply:
Actually, the `unwrap` isn't necessary (29868983ec836a7828c18b780826dabf71ca455d) but the assumptions remain.

Reply:
> > which requires an `unwrap` in `InvoiceRequestContentsWithoutPayerId::metadata`: [d931c89](https://github.com/lightningdevkit/rust-lightning/commit/d931c890cc1527ee6d9690c8e5b788f1d20ebdd3).
> 
> Seems like we can do a (maybe cleaned up) version of this:
> 
> ```
> diff --git a/lightning/src/offers/invoice_request.rs b/lightning/src/offers/invoice_request.rs
> index b64c53ad3..a845bbab3 100644
> --- a/lightning/src/offers/invoice_request.rs
> +++ b/lightning/src/offers/invoice_request.rs
> @@ -485,6 +485,7 @@ impl InvoiceRequestContents {
> 
>  impl InvoiceRequestContentsWithoutPayerId {
>         pub fn metadata(&self) -> &[u8] {
> +               if self.payer.0.as_bytes().is_none() { return &[] }
>                 &self.payer.0.as_bytes().unwrap()[..]
>         }
> ```
> 

Heh, actually had done the same thing for the offer metadata in the follow-up PR in a different context lol.

> That said, I don't have a super strong opinion based on those diffs. It's nice to unify the field but if you think the code is nicer without, that sg. If we keep the PR as-is then I'd rather rename `metadata` to `metadata_strategy` or similar.

Ended up going with the unified approach.